src/utils.rs

```diff
@@ -212,3 +212,28 @@ pub fn format_pairing_check_values(
 
     Ok([(msg_hash_arr, pk_arr), (sig_arr, n_g2)])
 }
+
+pub fn format_pairing_check_uncompressed_values(
+    message: Vec<u8>,
+    mut signature: Vec<u8>,
+    mut public_key: Vec<u8>,
+) -> Result<[([u8; 64], [u8; 128]); 2]> {
+    // convert to little endian
+    for i in (0..=32).step_by(32) {
+        signature[i..i + 32].reverse()
+    }
+    for i in (0..=96).step_by(32) {
+        public_key[i..i + 32].reverse()
+    }
```

Comment:
Dumb question time :D I see we're using `byteorder::BigEndian` in this file, is `byteorder::LittleEndian` an option or is that slower than doing it manually or just plain wrong?

Reply:
Good question!
I don’t think byteorder::LittleEndian is an option here as we’re receiving signatures and public keys as vectors of bytes not slices of integers, hence we can’t use byteorder::LittleEndian::write_u32/read_u32 for example.